src/main/java/reposense/model/RepoLocation.java

```diff
@@ -80,10 +103,16 @@ public boolean isEmpty() {
     }
 
     public String getRepoName() {
+        if (this.isLocal && !isEmpty()) {
+            return localRepoName;
+        }
         return repoName;
     }
 
     public String getOrganization() {
+        if (this.isLocal && !isEmpty()) {
+            return localOrganization;
+        }
```

Comment:
I was thinking that instead of storing `transient` fields `localRepoName` and `localOrganization`, we instead store `repoName` and `organization` as transient fields. Then change `localRepoName` and `localOrganization` to `remoteRepoName` and `remoteOrganization` which will be the ones uploaded via the JSON file. This way, these 2 methods do not need the `if-else` and `isLocal` field since we can always just return `repoName` and `organization`.

Reply:
That's probably a good idea. I was trying my best not to have to change anything outside of this class, but your suggestion seems cleaner overall.

Reply:
Oops yea. I believe you might need to change the key references in `api.js` as well.

Reply:
What do you think of the current solution?

Reply:
I think it looks good! I'll approve the PR once you've added the logger warning.